diff -ruNbB BoxLib-16.04/Tools/Py_util/GNUmakefile BoxLib-16.04-new/Tools/Py_util/GNUmakefile
--- BoxLib-16.04/Tools/Py_util/GNUmakefile	2020-06-19 18:24:27.000000000 -0600
+++ BoxLib-16.04-new/Tools/Py_util/GNUmakefile	2020-06-19 18:25:20.000000000 -0600
@@ -4,6 +4,9 @@
 MKVERBOSE :=t 
 
 COMP := gfortran
+ifdef BOXLIB_f2py_f90
+  COMP=$(BOXLIB_f2py_f90)
+endif
 
 
 # include the main Makefile stuff
@@ -31,7 +34,7 @@
 all: python_module
 
 python_module: $(objects)
-	f2py --fcompiler=$(F90) --f90flags="-J $(mdir)" -c fsnapshot.f90 -m fsnapshot $(objects) -lstdc++
+	f2py --f90exec=$(BOXLIB_f2py_f90) --f90flags="-J $(mdir)" -c fsnapshot.f90 -m fsnapshot $(objects) -lstdc++
 
 include $(BOXLIB_HOME)/Tools/F_mk/GMakerules.mak
 
