diff -ruNbB alquimia-dev-1.0.4/alquimia/pflotran_alquimia_interface.F90 alquimia-1.0.4-source/alquimia/pflotran_alquimia_interface.F90
--- alquimia-dev-1.0.4/alquimia/pflotran_alquimia_interface.F90	2017-01-11 19:35:52.000000000 -0700
+++ alquimia-1.0.4-source/alquimia/pflotran_alquimia_interface.F90	2017-09-01 16:56:06.120033584 -0600
@@ -64,7 +64,7 @@
   use Reactive_Transport_Aux_module, only : reactive_transport_auxvar_type
   use Global_Aux_module, only : global_auxvar_type
   use Material_Aux_class, only : material_auxvar_type
-  use Constraint_module, only : tran_constraint_list_type, tran_constraint_coupler_type
+  use Transport_Constraint_module, only : tran_constraint_list_type, tran_constraint_coupler_type
 
   use PFLOTRAN_Constants_module
 
@@ -151,12 +151,13 @@
   use Material_Aux_class, only : material_auxvar_type, MaterialAuxVarInit
   use Option_module, only : option_type, OptionCreate
   use Input_Aux_module, only : input_type, InputCreate, InputDestroy
-  use Constraint_module, only : tran_constraint_list_type, &
+  use Transport_Constraint_module, only : tran_constraint_list_type, &
        tran_constraint_coupler_type, TranConstraintCouplerCreate
 
-  implicit none
+#include "petsc/finclude/petscsys.h"
+  use petscsys 
 
-#include "finclude/petscsys.h"
+  implicit none
 
   ! function parameters
   character(kind=c_char), dimension(*), intent(in) :: input_filename
@@ -288,7 +289,7 @@
   ! pflotran
   use Option_module, only : OptionDestroy
   use Reaction_aux_module, only : ReactionDestroy
-  use Constraint_module, only : TranConstraintCouplerDestroy, TranConstraintDestroyList
+  use Transport_Constraint_module, only : TranConstraintCouplerDestroy, TranConstraintDestroyList
 
   implicit none
 
@@ -338,12 +339,13 @@
 
   ! pflotran
   use String_module, only : StringCompareIgnoreCase
-  use Constraint_module, only : tran_constraint_type, TranConstraintAddToList
+  use Transport_Constraint_module, only : tran_constraint_type, TranConstraintAddToList
   use Option_module, only : printMsg
 
-  implicit none
+#include "petsc/finclude/petscsys.h"
+  use petscsys 
 
-#include "finclude/petscsys.h"
+  implicit none
 
   ! function parameters
   type (c_ptr), intent(inout) :: pft_engine_state
@@ -559,7 +561,7 @@
   use AlquimiaContainers_module
 
   ! pflotran
-  use Mineral_module, only : RMineralSaturationIndex
+  use Reaction_Mineral_module, only : RMineralSaturationIndex
 
   implicit none
 
@@ -839,9 +841,10 @@
 
   use Option_module, only : option_type
 
-  implicit none
+#include "petsc/finclude/petscsys.h"
+  use petscsys 
 
-#include "finclude/petscsys.h"
+  implicit none
 
   ! function parameters
   character(kind=c_char), dimension(*), intent(in) :: input_filename
@@ -956,9 +959,10 @@
        InputCheckExit, InputError, InputErrorMsg, InputReadStringErrorMsg
   use String_module, only : StringToUpper
 
-  implicit none
+#include "petsc/finclude/petscsys.h"
+  use petscsys
 
-#include "finclude/petscsys.h"
+  implicit none
 
   ! function parameters
   type(option_type), pointer, intent(inout) :: option
@@ -1017,9 +1021,10 @@
   use Option_module, only : option_type
   use Input_Aux_module, only : input_type, InputFindStringInFile, InputError
 
-  implicit none
+#include "petsc/finclude/petscsys.h"
+  use petscsys
 
-#include "finclude/petscsys.h"
+  implicit none
 
   ! function parameters
   type(option_type), pointer, intent(inout) :: option
@@ -1048,13 +1053,14 @@
   ! pflotran
   use Reaction_module, only : ReactionInit, ReactionReadPass2
   use Reaction_Aux_module, only : reaction_type, ACT_COEF_FREQUENCY_OFF
-  use Database_module, only : DatabaseRead, BasisInit
+  use Reaction_Database_module, only : DatabaseRead, BasisInit
   use Option_module, only : option_type
   use Input_Aux_module, only : input_type, InputFindStringInFile, InputError
 
-  implicit none
+#include "petsc/finclude/petscsys.h"
+  use petscsys
 
-#include "finclude/petscsys.h"
+  implicit none
 
   ! function parameters
   type(option_type), pointer, intent(in) :: option
@@ -1108,13 +1114,14 @@
   use Input_Aux_module, only : input_type, InputReadPflotranString, InputReadWord, &
        InputErrorMsg, InputError
   use String_module, only : StringToUpper
-  use Constraint_module, only : tran_constraint_list_type, tran_constraint_type, &
+  use Transport_Constraint_module, only : tran_constraint_list_type, tran_constraint_type, &
        TranConstraintRead, TranConstraintInitList, TranConstraintAddToList, &
        TranConstraintCreate
 
-  implicit none
+#include "petsc/finclude/petscsys.h"
+  use petscsys 
 
-#include "finclude/petscsys.h"
+  implicit none
 
   ! function parameters
   type(option_type), pointer, intent(in) :: option
@@ -1180,12 +1187,13 @@
   use Reactive_Transport_Aux_module, only : reactive_transport_auxvar_type
   use Global_Aux_module, only : global_auxvar_type
   use Material_Aux_class, only : material_auxvar_type
-  use Constraint_module, only : tran_constraint_type, tran_constraint_coupler_type
+  use Transport_Constraint_module, only : tran_constraint_type, tran_constraint_coupler_type
   use Option_module, only : option_type, printMsg
 
-  implicit none
+#include "petsc/finclude/petscsys.h"
+  use petscsys
 
-#include "finclude/petscsys.h"
+  implicit none
 
   ! function parameters
   type(option_type), pointer, intent(in) :: option
@@ -1269,18 +1277,18 @@
   use Option_module, only : option_type, printErrMsg, printMsg
   use Reaction_aux_module, only : reaction_type, aq_species_constraint_type, &
        AqueousSpeciesConstraintCreate
-  use Mineral_aux_module, only : mineral_constraint_type, MineralConstraintCreate
+  use Reaction_Mineral_aux_module, only : mineral_constraint_type, MineralConstraintCreate
   use String_module, only : StringCompareIgnoreCase
-  use Constraint_module, only : tran_constraint_type, TranConstraintCreate, &
+  use Transport_Constraint_module, only : tran_constraint_type, TranConstraintCreate, &
        CONSTRAINT_FREE, CONSTRAINT_TOTAL, CONSTRAINT_TOTAL_SORB, &
        CONSTRAINT_PH, CONSTRAINT_MINERAL, &
        CONSTRAINT_GAS, CONSTRAINT_CHARGE_BAL
 
+#include "petsc/finclude/petscsys.h"
+  use petscsys
 
   implicit none
 
-#include "finclude/petscsys.h"
-
   ! function parameters
   type(option_type), pointer, intent(in) :: option
   type(reaction_type), pointer, intent(in) :: reaction
@@ -1530,7 +1538,7 @@
   call c_f_pointer(prop%mineral_rate_cnst%data, data, &
        (/prop%mineral_rate_cnst%size/))
   do i = 1, prop%mineral_rate_cnst%size
-     reaction%mineral%kinmnrl_rate(i) = data(i)
+     reaction%mineral%kinmnrl_rate_constant(i) = data(i)
   end do
 
   !
@@ -1956,11 +1964,12 @@
 ! **************************************************************************** !
 subroutine PrintTranConstraint(tran_constraint)
 
-  use Constraint_module, only : tran_constraint_type
+  use Transport_Constraint_module, only : tran_constraint_type
 
-  implicit none
+#include "petsc/finclude/petscsys.h"
+  use petscsys
 
-#include "finclude/petscsys.h"
+  implicit none
 
   ! function parameters
   type (tran_constraint_type), pointer :: tran_constraint
@@ -2004,7 +2013,7 @@
 
 subroutine PrintMineralConstraint(minerals)
 
-  use Mineral_aux_module, only : mineral_constraint_type
+  use Reaction_Mineral_aux_module, only : mineral_constraint_type
 
   implicit none
 
diff -ruNbB alquimia-dev-1.0.4/CMakeLists.txt alquimia-1.0.4-source/CMakeLists.txt
--- alquimia-dev-1.0.4/CMakeLists.txt	2017-01-11 19:35:52.000000000 -0700
+++ alquimia-1.0.4-source/CMakeLists.txt	2017-09-01 16:02:06.471814566 -0600
@@ -27,24 +27,24 @@
 
 # Make sure compilers are set. This must be done before enabling languages.
 if (NOT CMAKE_C_COMPILER)
-  if (NOT $ENV{CC} STREQUAL "")
-    set(CMAKE_C_COMPILER $ENV{CC})
+  if (NOT ${CC} STREQUAL "")
+    set(CMAKE_C_COMPILER ${CC})
   else()
     set(CMAKE_C_COMPILER cc)
   endif()
 endif()
 if (NOT CMAKE_C_FLAGS)
-  set(CMAKE_C_FLAGS $ENV{CFLAGS})
+  set(CMAKE_C_FLAGS ${CFLAGS})
 endif()
 if (NOT CMAKE_Fortran_COMPILER)
-  if (NOT $ENV{FC} STREQUAL "")
-    set(CMAKE_Fortran_COMPILER $ENV{FC})
+  if (NOT ${FC} STREQUAL "")
+    set(CMAKE_Fortran_COMPILER ${FC})
   else()
     set(CMAKE_Fortran_COMPILER gfortran)
   endif()
 endif()
 if (NOT CMAKE_Fortran_FLAGS)
-  set(CMAKE_Fortran_FLAGS $ENV{FCFLAGS})
+  set(CMAKE_Fortran_FLAGS ${FCFLAGS})
 endif()
 enable_language(C)
 enable_language(Fortran)
@@ -185,14 +185,14 @@
 
 # If needed, Check for PETSc and set things up.
 if (ALQUIMIA_NEED_PETSC)
-  if ($ENV{PETSC_DIR} STREQUAL "")
+  if (${PETSC_DIR} STREQUAL "")
     message(FATAL_ERROR "PETSC_DIR must be set for the requested engines to be enabled.")
   endif()
-  if ($ENV{PETSC_ARCH} STREQUAL "")
+  if (${PETSC_ARCH} STREQUAL "")
     message(FATAL_ERROR "PETSC_ARCH must be set for the requested engines to be enabled.")
   endif()
-  set(PETSC_DIR $ENV{PETSC_DIR})
-  set(PETSC_ARCH $ENV{PETSC_ARCH})
+  set(PETSC_DIR ${PETSC_DIR})
+  set(PETSC_ARCH ${PETSC_ARCH})
 
   # Do we have settings for PETSc's libraries/includes?
   if (TPL_PETSC_INCLUDE_DIRS)
