diff -Naur ascem-io-2.2/src/parallelIOwrite.c ascemio-2.2-source/src/parallelIOwrite.c
--- ascem-io-2.2/src/parallelIOwrite.c	2012-08-29 13:57:16.000000000 -0600
+++ ascemio-2.2-source/src/parallelIOwrite.c	2018-01-15 22:38:24.147239150 -0700
@@ -841,7 +841,7 @@
 		PRINT_MSG((ASCEMIO_VERBOSE, "Maximum string length: %d ", maxstr_len));
 
 		currfile = myIOgroup->file[fhandle];
-		buffer = (char *) malloc( nstrings * maxstr_len * sizeof(char));
+                buffer = (char *) malloc( nstrings * maxstr_len * sizeof(char) + 1);
 		pbuffer = buffer;
 
 		for (i=0; i < nstrings; i++)
