diff -ruNbB BoxLib-16.04/Src/CMakeLists.txt BoxLib-16.04-new/Src/CMakeLists.txt
--- BoxLib-16.04/Src/CMakeLists.txt	2016-04-11 16:42:12.000000000 -0600
+++ BoxLib-16.04-new/Src/CMakeLists.txt	2020-06-19 18:15:50.000000000 -0600
@@ -44,8 +44,14 @@
 SET_TARGET_PROPERTIES(cboxlib PROPERTIES LINKER_LANGUAGE C)
 SET_TARGET_PROPERTIES(cfboxlib PROPERTIES LINKER_LANGUAGE C)
 SET_TARGET_PROPERTIES(fboxlib PROPERTIES LINKER_LANGUAGE C)
+if (APPLE)
+  set_target_properties(cboxlib PROPERTIES LINK_FLAGS "-Wl,-undefined,dynamic_lookup")
+  set_target_properties(cfboxlib PROPERTIES LINK_FLAGS "-Wl,-undefined,dynamic_lookup")
+  set_target_properties(fboxlib PROPERTIES LINK_FLAGS "-Wl,-undefined,dynamic_lookup")
+endif()
 
 add_subdirectory(${CAMRDATADIR})
 
 add_subdirectory(Extern/gslib)
 
+add_dependencies(box_f box_c)
diff -ruNbB BoxLib-16.04/Src/C_BaseLib/FArrayBox.cpp BoxLib-16.04-new/Src/C_BaseLib/FArrayBox.cpp
--- BoxLib-16.04/Src/C_BaseLib/FArrayBox.cpp	2016-04-11 16:42:12.000000000 -0600
+++ BoxLib-16.04-new/Src/C_BaseLib/FArrayBox.cpp	2020-06-19 18:15:50.000000000 -0600
@@ -21,10 +21,10 @@
 #include <BL_CXX11.H>
 #include <MemPool.H>

-#if !defined(__PGI) && (__GNUC__ >= 5 || defined(BL_Darwin))
+// #if !defined(__PGI) && (__GNUC__ >= 5 || defined(BL_Darwin))
 using std::isinf;
 using std::isnan;
-#endif
+// #endif

 bool FArrayBox::initialized = false;

diff -ruNbB BoxLib-16.04/Src/Extern/amrdata/CMakeLists.txt BoxLib-16.04-new/Src/Extern/amrdata/CMakeLists.txt
--- BoxLib-16.04/Src/Extern/amrdata/CMakeLists.txt	2016-04-11 16:42:12.000000000 -0600
+++ BoxLib-16.04-new/Src/Extern/amrdata/CMakeLists.txt	2020-06-19 18:15:50.000000000 -0600
@@ -28,6 +28,9 @@
 set(local_source_files ${CXX_source_files} ${F77_source_files} ${FPP_out_files} ${F90_source_files})
 set(local_header_files ${CXX_header_files} ${F77_header_files} ${FPP_header_files} ${F90_header_files})
 add_library(box_camrdata ${local_source_files})
+if (APPLE)
+  set_target_properties(box_camrdata PROPERTIES LINK_FLAGS "-Wl,-undefined,dynamic_lookup")
+endif()
 
 add_install_include_file(${local_header_files})
 add_install_library(box_camrdata)
diff -ruNbB BoxLib-16.04/Src/LinearSolvers/C_CellMG/CGSolver.cpp BoxLib-16.04-new/Src/LinearSolvers/C_CellMG/CGSolver.cpp
--- BoxLib-16.04/Src/LinearSolvers/C_CellMG/CGSolver.cpp	2016-04-11 16:42:12.000000000 -0600
+++ BoxLib-16.04-new/Src/LinearSolvers/C_CellMG/CGSolver.cpp	2020-06-19 18:15:50.000000000 -0600
@@ -4,9 +4,9 @@
 #include <iomanip>
 #include <cmath>
 
-#if (__GNUC__ >= 6 || defined(BL_Darwin))
+// #if (__GNUC__ >= 6 || defined(BL_Darwin))
 using std::isinf;
-#endif
+// #endif
 
 #include <ParmParse.H>
 #include <ParallelDescriptor.H>
diff -ruNbB BoxLib-16.04/Tools/CMake/PreprocessBoxLibFortran.cmake BoxLib-16.04-new/Tools/CMake/PreprocessBoxLibFortran.cmake
--- BoxLib-16.04/Tools/CMake/PreprocessBoxLibFortran.cmake	2016-04-11 16:42:12.000000000 -0600
+++ BoxLib-16.04-new/Tools/CMake/PreprocessBoxLibFortran.cmake	2020-06-19 18:15:50.000000000 -0600
@@ -66,11 +66,13 @@
         list(APPEND PREPROCESS_FLAGS "-traditional")
       elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
         list(APPEND PREPROCESS_FLAGS "-traditional")
+      elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
+        list(APPEND PREPROCESS_FLAGS "-extend-source")
       endif()
 
       add_custom_command(
         OUTPUT ${of}
-        COMMAND ${CMAKE_C_COMPILER} -E ${PREPROCESS_FLAGS} ${inclflags} ${defflags} ${f} |
+	COMMAND ${CMAKE_Fortran_COMPILER} -E ${PREPROCESS_FLAGS} ${inclflags} ${defflags} ${f} |
                  ${PERL_EXECUTABLE} ${CCSE_PERL_DIR}/strip72 -c > ${of}
         DEPENDS ${f}
         COMMENT "Preprocessing ${f}..."
