--- ../BoxLib-16.10/Src/C_BaseLib/bl_fort_module.F90	2016-10-03 09:54:09.000000000 -0600
+++ ccse-16.10-source/Src/C_BaseLib/bl_fort_module.F90	2020-06-20 13:11:07.000000000 -0600
@@ -1,15 +1,15 @@
-module bl_fort_module
+      module bl_fort_module
 
-  use iso_c_binding, only : c_float, c_double
+      use iso_c_binding, only : c_float, c_double
 
-  implicit none
+      implicit none
 
-  integer, parameter :: bl_spacedim = BL_SPACEDIM
+      integer, parameter :: bl_spacedim = BL_SPACEDIM
 
 #ifdef BL_USE_FLOAT
-  integer, parameter :: c_real = c_float
+      integer, parameter :: c_real = c_float
 #else
-  integer, parameter :: c_real = c_double
+      integer, parameter :: c_real = c_double
 #endif
 
-end module bl_fort_module
+      end module bl_fort_module
