--- BoxLib-16.11/Src/C_BaseLib/CMakeLists.txt	2016-11-01 10:17:51.000000000 -0600
+++ BoxLib-16.11-new/Src/C_BaseLib/CMakeLists.txt	2020-06-22 01:19:46.000000000 -0600
@@ -27,7 +27,7 @@
 set(F90_header_files)

 preprocess_boxlib_fortran(FPP_out_files ${FPP_source_files})
-preprocess_boxlib_fortran90(F90PP_out_files ${F90PP_source_files})
+preprocess_boxlib_fortran(F90PP_out_files ${F90PP_source_files})
 set(local_source_files ${CXX_source_files} ${F77_source_files} ${FPP_out_files} ${F90PP_out_files} ${F90_source_files})
 set(local_header_files ${CXX_header_files} ${F77_header_files} ${FPP_header_files} ${F90_header_files})
 add_library(box_c OBJECT ${local_source_files})

