diff -ruNbB ccse-orig/Src/F_BaseLib/CMakeLists.txt ccse-new/Src/F_BaseLib/CMakeLists.txt
--- ccse-orig/Src/F_BaseLib/CMakeLists.txt	2015-09-24 18:01:31.000000000 -0600
+++ ccse-new/Src/F_BaseLib/CMakeLists.txt	2020-05-02 13:29:57.000000000 -0600
@@ -26,14 +26,13 @@
 endif()
 
 if (ENABLE_BACKTRACE)
-  list(APPEND F90_source_files bl_prof_backtrace.f90)
-else ()
+  list(APPEND F90_source_files backtrace_f.f90)
+endif()
   if (ENABLE_PROFILING)
     list(APPEND F90_source_files bl_prof.f90)
   else()
     list(APPEND F90_source_files bl_prof_stubs.f90)
   endif()
-endif()
 
 if (ENABLE_OMP)
   list(APPEND F90_source_files omp.f90)
