diff -ruNbB ccse-16.03-source-orig/Src/C_BaseLib/FabArray.H ccse-16.03-source/Src/C_BaseLib/FabArray.H
--- ccse-16.03-source-orig/Src/C_BaseLib/FabArray.H	2020-06-15 20:58:18.000000000 -0600
+++ ccse-16.03-source/Src/C_BaseLib/FabArray.H	2020-06-15 21:01:18.000000000 -0600
@@ -513,6 +513,8 @@
 
     void flushFPinfo ();
 
+    const Array<int>& IndexMap () const { return indexMap; }
+
 protected:
 
     DistributionMapping& ModifyDistributionMap () { return distributionMap; }
@@ -520,7 +522,7 @@
     //
     // Returns constant reference to indices in the FabArray that we have access
     //
-    const Array<int>& IndexMap () const { return indexMap; }
+    // const Array<int>& IndexMap () const { return indexMap; }
     //
     // Returns owenership of fabs. The concept of ownership only applies when UPC++
     // team is used. In that case, each fab is shared by team workers, with one
