diff -ruNbB BoxLib-16.04/Src/F_BaseLib/CMakeLists.txt BoxLib-16.04-new/Src/F_BaseLib/CMakeLists.txt
--- BoxLib-16.04/Src/F_BaseLib/CMakeLists.txt	2020-06-19 19:11:19.000000000 -0600
+++ BoxLib-16.04-new/Src/F_BaseLib/CMakeLists.txt	2020-06-19 19:09:45.000000000 -0600
@@ -11,7 +11,7 @@
 # through these variables.
 project(FBOXLIB)
 
-include_directories(${FBOXLIB_INCLUDE_DIRS})
+include_directories(${FBOXLIB_INCLUDE_DIRS} ${CBOXLIB_INCLUDE_DIRS})
 
 set(CXX_source_files backtrace_c.cpp)
 set(C_source_files fabio_c.c  ppm_util_c.c  system_util_c.c  timer_c.c)
