diff -Naur ccse-1.3.4-old/Tools/Py_util/GNUmakefile ccse-1.3.4-new/Tools/Py_util/GNUmakefile
--- ccse-1.3.4-old/Tools/Py_util/GNUmakefile	2015-09-24 18:01:31.000000000 -0600
+++ ccse-1.3.4-new/Tools/Py_util/GNUmakefile	2017-09-13 08:57:49.000000000 -0600
@@ -4,6 +4,9 @@
 MKVERBOSE :=t 
 
 COMP := gfortran
+ifdef BOXLIB_f2py_f90
+  COMP=$(BOXLIB_f2py_f90)
+endif
 
 
 # include the main Makefile stuff
@@ -31,7 +34,7 @@
 all: python_module
 
 python_module: $(objects)
-	f2py --f90flags="-J $(mdir)" -c fsnapshot.f90 -m fsnapshot $(objects) -lstdc++
+	f2py --f90exec=$(BOXLIB_f2py_f90) --f90flags="-J $(mdir)" -c fsnapshot.f90 -m fsnapshot $(objects) -lstdc++
 
 include $(BOXLIB_HOME)/Tools/F_mk/GMakerules.mak
 
