diff -ruNbB ccse-16.03-source-orig/Src/C_BaseLib/FabArray.H ccse-16.03-source/Src/C_BaseLib/FabArray.H
--- ccse-16.03-source-orig/Src/C_BaseLib/FabArray.H	2020-06-15 20:58:18.000000000 -0600
+++ ccse-16.03-source/Src/C_BaseLib/FabArray.H	2020-06-15 21:01:18.000000000 -0600
@@ -420,6 +420,8 @@
 
     const TileArray* getTileArray (const IntVect& tilesize) const;
 
+    const Array<int>& IndexMap () const { return indexMap; }
+
 protected:
 
     DistributionMapping& ModifyDistributionMap () { return distributionMap; }
@@ -427,7 +429,7 @@
     //
     // Returns constant reference to indices in the FabArray that we own.
     //
-    const Array<int>& IndexMap () const { return indexMap; }
+    // const Array<int>& IndexMap () const { return indexMap; }
     //
     // Returns local index in the vector of FABs.
     //
