diff --color -ruN coolprop-7.0.0-orig/CMakeLists.txt coolprop-7.0.0-source/CMakeLists.txt
--- coolprop-7.0.0-orig/CMakeLists.txt	2025-10-25 18:14:46
+++ coolprop-7.0.0-source/CMakeLists.txt	2025-10-25 18:09:16
@@ -537,8 +537,9 @@
     "src/CoolPropLib.cpp"
     CACHE STRING "The file that contains the exported functions")
 
+file(GLOB COOLPROP_LIBRARY_HEADER_TMP RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} "include/*.h")
 set(COOLPROP_LIBRARY_HEADER
-    "include/CoolPropLib.h"
+    ${COOLPROP_LIBRARY_HEADER_TMP}
     CACHE STRING "The file that contains the export header")
 
 set(COOLPROP_LIBRARY_NAME
@@ -620,10 +621,10 @@
 	
     install(
       TARGETS ${LIB_NAME}
-      DESTINATION ${OUTPUT_FOLDER}
+      DESTINATION ${COOLPROP_INSTALL_PREFIX}/lib
     )
     install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/${COOLPROP_LIBRARY_HEADER}
-            DESTINATION shared_library)
+            DESTINATION ${COOLPROP_INSTALL_PREFIX}/include)
     set_property(
       TARGET ${LIB_NAME}
       APPEND_STRING
