diff -ruNbB crunchtope-160915.c31ecb9.0-source-orig/AssembleGlobal.F90 crunchtope-160915.c31ecb9.0-source/AssembleGlobal.F90
--- crunchtope-160915.c31ecb9.0-source-orig/AssembleGlobal.F90	2017-08-14 23:04:39.873152793 -0600
+++ crunchtope-160915.c31ecb9.0-source/AssembleGlobal.F90	2017-08-14 23:36:33.423081299 -0600
@@ -60,7 +60,7 @@
 
 !*****************************PETSc include statements ********************
 
-#include "petsc/finclude/petsc.h"
+#include "finclude/petsc.h"
 
 !**************************** End PETSc include statements **************
 
diff -ruNbB crunchtope-160915.c31ecb9.0-source-orig/CMakeLists.txt crunchtope-160915.c31ecb9.0-source/CMakeLists.txt
--- crunchtope-160915.c31ecb9.0-source-orig/CMakeLists.txt	1969-12-31 17:00:00.000000000 -0700
+++ crunchtope-160915.c31ecb9.0-source/CMakeLists.txt	2017-08-14 23:54:24.212453529 -0600
@@ -0,0 +1,40 @@
+# Minimum CMake version.
+cmake_minimum_required (VERSION 2.8.12)
+set(CMAKE_BUILD_TYPE Release)
+
+project(crunchtope)
+
+# Options for building CrunchTope. These come from the xSDK compliance rules.
+option(USE_XSDK_DEFAULTS "Set to use xSDK defaults for options [ON]." ON)
+option(CMAKE_INSTALL_PREFIX "Sets installation prefix [/usr/local].")
+option(XSDK_ENABLE_DEBUG "Enables Debug mode builds [OFF]." OFF)
+option(BUILD_SHARED_LIBS "Builds shared libraries [ON]." ON)
+
+# For now, we disable shared libs on Macs.
+if (APPLE)
+  set(BUILD_SHARED_LIBS OFF)
+endif()
+
+# Make sure compilers are set. This must be done before enabling languages.
+if (NOT CMAKE_Fortran_COMPILER)
+  message(FATAL_ERROR "Fortran compiler is not set")
+endif()
+enable_language(Fortran)
+
+set(CMAKE_Fortran_FLAGS_RELEASE "${CMAKE_Fortran_FLAGS_RELEASE} -w -DALQUIMIA -Wall -Wno-unused-variable -ffree-line-length-0")
+
+message(STATUS "C compiler is ${CMAKE_C_COMPILER} (${CMAKE_C_COMPILER_ID})")
+message(STATUS "Fortran compiler is ${CMAKE_Fortran_COMPILER} (${CMAKE_Fortran_COMPILER_ID})")
+
+# Include the binary directory in the header file search path,
+# since it's where we place generated files.
+include_directories("${PROJECT_SOURCE_DIR}")
+
+# Now that we have gathered all tech information, we create a library
+add_definitions(-DALQUIMIA)
+file(GLOB CRUNCH_SRC "*.F90")
+
+include_directories(${PETSC_DIR}/include)
+add_library(${CRUNCHTOPE_LIB} STATIC ${CRUNCH_SRC})
+
+
diff -ruNbB crunchtope-160915.c31ecb9.0-source-orig/CrunchPETScFinalizeSolver.F90 crunchtope-160915.c31ecb9.0-source/CrunchPETScFinalizeSolver.F90
--- crunchtope-160915.c31ecb9.0-source-orig/CrunchPETScFinalizeSolver.F90	2017-08-14 23:04:39.876486208 -0600
+++ crunchtope-160915.c31ecb9.0-source/CrunchPETScFinalizeSolver.F90	2017-08-14 23:49:46.775708439 -0600
@@ -47,7 +47,7 @@
 
 IMPLICIT NONE
 
-#include "petsc/finclude/petsc.h"
+#include "finclude/petsc.h"
 
 !  External variables and arrays
 
diff -ruNbB crunchtope-160915.c31ecb9.0-source-orig/CrunchPETScInitializeChemistry.F90 crunchtope-160915.c31ecb9.0-source/CrunchPETScInitializeChemistry.F90
--- crunchtope-160915.c31ecb9.0-source-orig/CrunchPETScInitializeChemistry.F90	2017-08-14 23:04:39.876486208 -0600
+++ crunchtope-160915.c31ecb9.0-source/CrunchPETScInitializeChemistry.F90	2017-08-14 23:50:32.516820505 -0600
@@ -48,7 +48,7 @@
 
 IMPLICIT NONE
 
-#include "petsc/finclude/petsc.h"
+#include "finclude/petsc.h"
 
 !  External variables and arrays
 
diff -ruNbB crunchtope-160915.c31ecb9.0-source-orig/CrunchPETScInitializeDiffusion.F90 crunchtope-160915.c31ecb9.0-source/CrunchPETScInitializeDiffusion.F90
--- crunchtope-160915.c31ecb9.0-source-orig/CrunchPETScInitializeDiffusion.F90	2017-08-14 23:04:39.876486208 -0600
+++ crunchtope-160915.c31ecb9.0-source/CrunchPETScInitializeDiffusion.F90	2017-08-14 23:50:59.314138672 -0600
@@ -49,7 +49,7 @@
 
 IMPLICIT NONE
 
-#include "petsc/finclude/petsc.h"
+#include "finclude/petsc.h"
 
 
 !  External variables and arrays
diff -ruNbB crunchtope-160915.c31ecb9.0-source-orig/CrunchPETScInitializePressure.F90 crunchtope-160915.c31ecb9.0-source/CrunchPETScInitializePressure.F90
--- crunchtope-160915.c31ecb9.0-source-orig/CrunchPETScInitializePressure.F90	2017-08-14 23:04:39.876486208 -0600
+++ crunchtope-160915.c31ecb9.0-source/CrunchPETScInitializePressure.F90	2017-08-14 23:51:20.461319473 -0600
@@ -47,7 +47,7 @@
 
 IMPLICIT NONE
 
-#include "petsc/finclude/petsc.h"
+#include "finclude/petsc.h"
 
 !  External variables and arrays
 
diff -ruNbB crunchtope-160915.c31ecb9.0-source-orig/CrunchPETScTolerances.F90 crunchtope-160915.c31ecb9.0-source/CrunchPETScTolerances.F90
--- crunchtope-160915.c31ecb9.0-source-orig/CrunchPETScTolerances.F90	2017-08-14 23:04:39.876486208 -0600
+++ crunchtope-160915.c31ecb9.0-source/CrunchPETScTolerances.F90	2017-08-14 23:51:39.495115559 -0600
@@ -47,7 +47,7 @@
 USE solver, ONLY:  level, SolverMethod, PCMethod
 IMPLICIT NONE
 
-#include "petsc/finclude/petsc.h"
+#include "finclude/petsc.h"
 
 !  External variables and arrays
 REAL(DP), INTENT(IN)                                                  :: rtolksp
diff -ruNbB crunchtope-160915.c31ecb9.0-source-orig/CrunchTope.F90 crunchtope-160915.c31ecb9.0-source/CrunchTope.F90
--- crunchtope-160915.c31ecb9.0-source-orig/CrunchTope.F90	2017-08-14 23:04:39.876486208 -0600
+++ crunchtope-160915.c31ecb9.0-source/CrunchTope.F90	2017-08-14 23:52:04.262384372 -0600
@@ -81,7 +81,7 @@
 
 !  **********  PETSc include statements *********************************
 
-#include "petsc/finclude/petsc.h"
+#include "finclude/petsc.h"
 
 ! ******************* end PETSc include statements ***********************
 
diff -ruNbB crunchtope-160915.c31ecb9.0-source-orig/GIMRTCrunchPETScTolerances.F90 crunchtope-160915.c31ecb9.0-source/GIMRTCrunchPETScTolerances.F90
--- crunchtope-160915.c31ecb9.0-source-orig/GIMRTCrunchPETScTolerances.F90	2017-08-14 23:04:39.879819623 -0600
+++ crunchtope-160915.c31ecb9.0-source/GIMRTCrunchPETScTolerances.F90	2017-08-14 23:52:51.316861702 -0600
@@ -47,7 +47,7 @@
 USE solver, ONLY:  GIMRTlevel, GIMRT_SolverMethod, GIMRT_PCMethod
 IMPLICIT NONE
 
-#include "petsc/finclude/petsc.h"
+#include "finclude/petsc.h"
 
 !  External variables and arrays
 REAL(DP), INTENT(IN)                                                  :: rtolksp
diff -ruNbB crunchtope-160915.c31ecb9.0-source-orig/Makefile crunchtope-160915.c31ecb9.0-source/Makefile
--- crunchtope-160915.c31ecb9.0-source-orig/Makefile	2017-08-14 23:04:39.879819623 -0600
+++ crunchtope-160915.c31ecb9.0-source/Makefile	2017-08-14 23:36:33.423081299 -0600
@@ -625,11 +625,11 @@
 ## For PETSc 3.0X
 ##include ${PETSC_DIR}/conf/base
 ## For PETSc 3.2X
-##include ${PETSC_DIR}/conf/variables
-##include ${PETSC_DIR}/conf/rules
+include ${PETSC_DIR}/conf/variables
+include ${PETSC_DIR}/conf/rules
 ## For PETSc >3.6.0
-include ${PETSC_DIR}/lib/petsc/conf/variables
-include ${PETSC_DIR}/lib/petsc/conf/rules
+##include ${PETSC_DIR}/lib/petsc/conf/variables
+##include ${PETSC_DIR}/lib/petsc/conf/rules
 
 CrunchMain : ${OBJSF} chkopts
 	-${FLINKER} -o CrunchTope ${OBJSF} ${PETSC_FORTRAN_LIB} ${PETSC_LIB} ${FFLAGS}
diff -ruNbB crunchtope-160915.c31ecb9.0-source-orig/Pressure.F90 crunchtope-160915.c31ecb9.0-source/Pressure.F90
--- crunchtope-160915.c31ecb9.0-source-orig/Pressure.F90	2017-08-14 23:04:39.883153038 -0600
+++ crunchtope-160915.c31ecb9.0-source/Pressure.F90	2017-08-14 23:54:58.563288673 -0600
@@ -56,7 +56,7 @@
 
 !*****************************PETSc include statements ********************
 
-#include "petsc/finclude/petsc.h"
+#include "finclude/petsc.h"
 
 !**************************** End PETSc include statements **************
 
