--- curl-curl-7_56_1/CMakeLists.txt	2017-10-22 22:51:20.000000000 -0700
+++ curl-7.56.1-source/CMakeLists.txt	2017-12-12 07:22:29.799551000 -0800
@@ -360,6 +360,9 @@
   set(HAVE_LIBCRYPTO ON)
   set(HAVE_LIBSSL ON)
   list(APPEND CURL_LIBS ${OPENSSL_LIBRARIES})
+  if (HAVE_LIBDL)
+    list(APPEND CURL_LIBS dl)
+  endif()
   include_directories(${OPENSSL_INCLUDE_DIR})
   set(CMAKE_REQUIRED_INCLUDES ${OPENSSL_INCLUDE_DIR})
   check_include_file("openssl/crypto.h" HAVE_OPENSSL_CRYPTO_H)
