--- hdf5-1.10.6/config/cmake_ext_mod/HDFMacros.cmake	2019-12-17 14:14:09.000000000 -0800
+++ hdf5-1.10.6-new/config/cmake_ext_mod/HDFMacros.cmake	2020-03-25 21:12:10.256059664 -0700
@@ -379,17 +379,21 @@
   set (CMAKE_INSTALL_RPATH_USE_LINK_PATH  FALSE)
   # add the automatically determined parts of the RPATH
   # which point to directories outside the build tree to the install RPATH
-  set (CMAKE_BUILD_WITH_INSTALL_RPATH ON)
+  set (CMAKE_BUILD_WITH_INSTALL_RPATH OFF)
   if (APPLE)
-    set (CMAKE_INSTALL_NAME_DIR "@rpath")
-    set (CMAKE_INSTALL_RPATH
-        "@executable_path/../${${package_prefix}_INSTALL_LIB_DIR}"
-        "@executable_path/"
-        "@loader_path/../${${package_prefix}_INSTALL_LIB_DIR}"
-        "@loader_path/"
-    )
+    set (CMAKE_MACOSX_RPATH ON)
+    # set (CMAKE_INSTALL_NAME_DIR "@rpath")
+    # set (CMAKE_INSTALL_RPATH
+    #    "@executable_path/../${${package_prefix}_INSTALL_LIB_DIR}"
+    #    "@executable_path/"
+    #    "@loader_path/../${${package_prefix}_INSTALL_LIB_DIR}"
+    #    "@loader_path/"
+    #)
   else ()
     set (CMAKE_INSTALL_RPATH "\$ORIGIN/../${${package_prefix}_INSTALL_LIB_DIR}:\$ORIGIN/")
+    if (BUILD_STATIC_EXECS)
+       set (CMAKE_SKIP_INSTALL_RPATH ON)
+    endif ()
   endif ()
 
   if (DEFINED ADDITIONAL_CMAKE_PREFIX_PATH AND EXISTS "${ADDITIONAL_CMAKE_PREFIX_PATH}")
