diff -rupN hdf5-1.8.8-source/tools/lib/h5diff.c hdf5-1.8.8-source-fixed/tools/lib/h5diff.c
--- hdf5-1.8.8-source/tools/lib/h5diff.c	2011-11-07 15:11:35.000000000 -0700
+++ hdf5-1.8.8-source-fixed/tools/lib/h5diff.c	2013-12-18 08:06:44.534616740 -0700
@@ -644,7 +644,7 @@ hsize_t h5diff(const char *fname1,
     char         filenames[2][MAX_FILENAME];
     hsize_t      nfound = 0;
     int i;
-    //int i1, i2;
+    /* int i1, i2; */
     int l_ret;
     const char * obj1fullname = NULL;
     const char * obj2fullname = NULL;
