===================================================
diff -ruNbB hypre-2.10.0b-source-orig/src/parcsr_ls/par_amg_solve.c hypre-2.10.0b-source/src/parcsr_ls/par_amg_solve.c
--- hypre-2.10.0b-source-orig/src/parcsr_ls/par_amg_solve.c    2015-02-16 16:03:16.826066558 -0700
+++ hypre-2.10.0b-source/src/parcsr_ls/par_amg_solve.c    2015-02-16 15:59:59.921637467 -0700
@@ -160,7 +160,7 @@
     *    Compute initial fine-grid residual and print 
     *-----------------------------------------------------------------------*/
 
-   if (amg_print_level > 1 || amg_logging > 1)
+   if (amg_print_level > 1 || amg_logging > 1 || tol > 0)
    {
      if ( amg_logging > 1 ) {
         hypre_ParVectorCopy(F_array[0], Residual );
=========================================================
