diff --git a/src/parcsr_ls/dsuperlu.c b/src/parcsr_ls/dsuperlu.c
index 31bd21825..7e1387cb8 100644
--- a/src/parcsr_ls/dsuperlu.c
+++ b/src/parcsr_ls/dsuperlu.c
@@ -116,9 +116,9 @@ HYPRE_Int hypre_SLUDistSetup( HYPRE_Solver *solver, hypre_ParCSRMatrix *A, HYPRE
    dslu_data->dslu_options.DiagPivotThresh = 1.0;
    dslu_data->dslu_options.ReplaceTinyPivot = NO; */
 
-   dScalePermstructInit(global_num_rows, global_num_rows, &(dslu_data->dslu_ScalePermstruct));
+   ScalePermstructInit(global_num_rows, global_num_rows, &(dslu_data->dslu_ScalePermstruct));
 
-   dLUstructInit(global_num_rows, &(dslu_data->dslu_data_LU));
+   LUstructInit(global_num_rows, &(dslu_data->dslu_data_LU));
 
    PStatInit(&(dslu_data->dslu_data_stat));
 
@@ -161,9 +161,9 @@ HYPRE_Int hypre_SLUDistDestroy( void* solver)
 
    PStatFree(&(dslu_data->dslu_data_stat));
    Destroy_CompRowLoc_Matrix_dist(&(dslu_data->A_dslu));
-   dScalePermstructFree(&(dslu_data->dslu_ScalePermstruct));
-   dDestroy_LU(dslu_data->global_num_rows, &(dslu_data->dslu_data_grid), &(dslu_data->dslu_data_LU));
-   dLUstructFree(&(dslu_data->dslu_data_LU));
+   ScalePermstructFree(&(dslu_data->dslu_ScalePermstruct));
+   Destroy_LU(dslu_data->global_num_rows, &(dslu_data->dslu_data_grid), &(dslu_data->dslu_data_LU));
+   LUstructFree(&(dslu_data->dslu_data_LU));
    if (dslu_data->dslu_options.SolveInitialized)
    {
       dSolveFinalize(&(dslu_data->dslu_options), &(dslu_data->dslu_solve));
diff --git a/src/parcsr_ls/dsuperlu.h b/src/parcsr_ls/dsuperlu.h
index add1d2e09..a0fef214f 100644
--- a/src/parcsr_ls/dsuperlu.h
+++ b/src/parcsr_ls/dsuperlu.h
@@ -17,12 +17,12 @@ typedef struct
    HYPRE_BigInt global_num_rows;
    SuperMatrix A_dslu;
    HYPRE_Real *berr;
-   dLUstruct_t dslu_data_LU;
+   LUstruct_t dslu_data_LU;
    SuperLUStat_t dslu_data_stat;
    superlu_dist_options_t dslu_options;
    gridinfo_t dslu_data_grid;
-   dScalePermstruct_t dslu_ScalePermstruct;
-   dSOLVEstruct_t dslu_solve;
+   ScalePermstruct_t dslu_ScalePermstruct;
+   SOLVEstruct_t dslu_solve;
 }
 hypre_DSLUData;
 
