+add_subdirectory(src)
diff -ruNbB hypre-2.14.0/src/CMakeLists.txt hypre-2.14.0-source/src/CMakeLists.txt
--- hypre-2.14.0/src/CMakeLists.txt	2018-03-16 16:14:09.000000000 -0600
+++ hypre-2.14.0-source/src/CMakeLists.txt	2018-05-18 11:04:13.000000000 -0600
@@ -58,6 +58,11 @@
   set (HYPRE_NO_GLOBAL_PARTITION OFF CACHE BOOL "" FORCE)
 endif ()
 
+if (HYPRE_USING_TPL_SUPERLU)
+  include_directories(${SUPERLU_INCLUDE_DIR})
+  add_definitions(-DHAVE_SUPERLU)
+  add_definitions(-DHAVE_DSUPERLU)
+endif ()
 #   --with-examples         Build example codes.
 #   --with-fei              Use internal FEI routines.
 #   --with-superlu          Use internal SuperLU routines.
@@ -524,6 +529,7 @@
   parcsr_ls/ame.c
   parcsr_ls/par_restr.c
   parcsr_ls/par_lr_restr.c
+  parcsr_ls/superlu.c
 )
 
 # Headers and sources: struct_mv
@@ -827,6 +833,8 @@
 endif ()
 
 add_library (HYPRE ${HYPRE_SOURCES} ${FEI_LIBS})
+target_link_libraries(HYPRE ${SUPERLU_LIBRARY})
+target_link_libraries(HYPRE ${SUPERLUDIST_LIBRARY})
 
 install (TARGETS HYPRE DESTINATION lib)
 install (FILES ${HYPRE_HEADERS} DESTINATION include)
