diff -ruNbB MSTK-3_0_3/cmake/modules/FindHDF5.cmake mstk-3.0.3-source/cmake/modules/FindHDF5.cmake
--- MSTK-3_0_3/cmake/modules/FindHDF5.cmake	2017-10-30 11:55:16.000000000 -0600
+++ mstk-3.0.3-source/cmake/modules/FindHDF5.cmake	2017-12-04 23:01:37.000000000 -0700
@@ -144,12 +144,13 @@
   set(_search_key "Extra libraries")
   set(_libraries)
   _HDF5_PARSE_SETTINGS_FILE(${_file} ${_search_key} _library_flags)
-  string( REGEX MATCHALL "[, ]-l([^\", ]+)|^-l([^\", ]+)" _library_name_flags ${_library_flags})
-  foreach ( _lib ${_library_name_flags} )
-    _HDF5_CHOMP_STRING(${_lib} _lib_chomp)
-    string( REGEX REPLACE "^[,]-l|^-l" "" _lib_chomp ${_lib_chomp})
-    list(APPEND _libraries ${_lib_chomp})
-  endforeach()
+  #string( REGEX MATCHALL "[, ]-l([^\", ]+)|^-l([^\", ]+)" _library_name_flags ${_library_flags})
+  #foreach ( _lib ${_library_name_flags} )
+  #  _HDF5_CHOMP_STRING(${_lib} _lib_chomp)
+  #  string( REGEX REPLACE "^[,]-l|^-l" "" _lib_chomp ${_lib_chomp})
+  #  list(APPEND _libraries ${_lib_chomp})
+  #endforeach()
+  string(REGEX REPLACE "^ " "" _libraries "${_library_flags}")
 
   # Grab all the extra library paths to build a search list
   _HDF5_EXTRA_LIBRARY_DIRS(${_file} _search_list)
