diff -uNbB mstk-3.1.0-orig/CMakeLists.txt mstk-3.1.0-source/CMakeLists.txt
--- mstk-3.1.0-orig/CMakeLists.txt	2019-02-22 13:00:08.000000000 -0700
+++ mstk-3.1.0-source/CMakeLists.txt	2019-02-22 13:00:19.000000000 -0700
@@ -91,7 +91,7 @@
 
  set(HDF5_ROOT ${HDF5_DIR})
  find_package(HDF5 REQUIRED 
-              COMPONENTS C CXX HL)
+              COMPONENTS C HL)
 
  if(NOT HDF5_FOUND)
     message(FATAL_ERROR "Error: could not find hdf5 library\n"
