diff -ruN mstk-3.3.5-source-old/CMakeLists.txt mstk-3.3.5-source/CMakeLists.txt
--- mstk-3.3.5-source-old/CMakeLists.txt	2020-08-12 20:37:45.000000000 -0400
+++ mstk-3.3.5-source/CMakeLists.txt	2020-08-12 20:48:38.000000000 -0400
@@ -61,6 +61,18 @@
 
 set (MSTKLIB mstk)
 
+# The main mstk target
+if (BUILD_SHARED_LIBS)
+  set(CMAKE_SKIP_BUILD_RPATH FALSE)
+  set(CMAKE_SKIP_INSTALL_RPATH FALSE)
+  set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
+  set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE) 
+  set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+  set(CMAKE_INSTALL_NAME_DIR "${CMAKE_INSTALL_RPATH}")
+else()
+  SET(CMAKE_SKIP_INSTALL_RPATH ON CACHE BOOL "Turn off for static install." FORCE)
+  SET(CMAKE_SKIP_RPATH ON CACHE BOOL "Turn off for static install." FORCE)
+endif()
 
 # Add the actual source directory - these will build up the list "mstkfiles"
 
