--- mstk-2.12-old/CMakeLists.txt	2014-05-07 13:24:53.000000000 -0700
+++ mstk-2.12-new/CMakeLists.txt	2014-05-20 22:17:02.358594630 -0700
@@ -22,8 +22,11 @@
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/modules/")
 
+if (PREFER_STATIC_LIBRARIES)
+   set(CMAKE_FIND_LIBRARY_SUFFIXES ".a;.so")
+endif()
 
-
+message(STATUS "JDM ----> CMAKE_FIND_LIBRARY_SUFFIXES=${CMAKE_FIND_LIBRARY_SUFFIXES}")
 
 ##############################################################################
 # MSTK LIBRARY BUILD CONFIGURATION
@@ -110,9 +113,6 @@
         DESTINATION ${INSTALL_DIR}/include)
 endif ()
 
-
-
-
 ################################################################################
 # Discover TPLs (Third Party Libraries) based on configure options
 ################################################################################
