diff -ruNbB mstk-3.0.1/CMakeLists.txt mstk-3.0.1-source/CMakeLists.txt
--- mstk-3.0.1/CMakeLists.txt	2017-06-23 15:03:58.000000000 -0600
+++ mstk-3.0.1-source/CMakeLists.txt	2017-10-21 22:32:21.000000000 -0600
@@ -335,6 +334,9 @@
 
 set(MSTK_TPL_LIBRARIES ${METIS_LIBRARIES} ${ZOLTAN_LIBRARIES} 
   ${ExodusII_LIBRARIES} ${NetCDF_LIBRARIES} ${HDF5_LIBRARIES})
+if (APPLE)
+target_link_libraries(${MSTKLIB} ${METIS_LIBRARIES} ${ZOLTAN_LIBRARIES} ${ExodusII_LIBRARIES} ${NetCDF_LIBRARIES} ${HDF5_LIBRARIES})
+endif()
 
 # Make a separate variables where the HDF5 libraries are not targets
 # but expanded library paths. This will be exported for use by other
 
 
