diff -ruNbB netcdf-4.6.1/CMakeLists.txt netcdf-4.6.1-source/CMakeLists.txt
--- netcdf-4.6.1/CMakeLists.txt	2018-03-15 15:22:52.000000000 -0600
+++ netcdf-4.6.1-source/CMakeLists.txt	2018-06-23 19:37:39.000000000 -0600
@@ -710,6 +714,7 @@
   # Use H5Pset_fapl_mpio and H5Pget_fapl_mpio, instead.
   # CHECK_LIBRARY_EXISTS(${HDF5_C_LIBRARY_hdf5} H5Pget_fapl_mpiposix "" HDF5_IS_PARALLEL_MPIPOSIX)
 
+  SET(CMAKE_REQUIRED_LIBRARIES "dl;${ZLIB_LIBRARY};m")
   CHECK_LIBRARY_EXISTS(${HDF5_C_LIBRARY_hdf5} H5Pget_fapl_mpio "" HDF5_IS_PARALLEL_MPIO)
   IF(HDF5_IS_PARALLEL_MPIO)
     SET(HDF5_PARALLEL ON)
diff -ruNbB netcdf-4.6.1/include/netcdf.h netcdf-4.6.1-source/include/netcdf.h
--- netcdf-4.7.0/include/netcdf.h	2018-03-15 15:22:57.000000000 -0600
+++ netcdf-4.7.0-source/include/netcdf.h	2018-06-23 19:39:22.000000000 -0600
@@ -269,11 +269,11 @@
       are *not* enforced after version 4.5.0
 */
 /**@{*/
-#define NC_MAX_DIMS     1024 /* not enforced after 4.5.0 */
+#define NC_MAX_DIMS    65536 /* not enforced after 4.5.0 */
 #define NC_MAX_ATTRS    8192 /* not enforced after 4.5.0 */
-#define NC_MAX_VARS     8192 /* not enforced after 4.5.0 */
+#define NC_MAX_VARS   524288 /* not enforced after 4.5.0 */
 #define NC_MAX_NAME     256
-#define NC_MAX_VAR_DIMS 1024 /**< max per variable dimensions */
+#define NC_MAX_VAR_DIMS  8 /**< max per variable dimensions */
 /**@}*/
 
 /** This is the max size of an SD dataset name in HDF4 (from HDF4 documentation).*/
