diff -ruNbB netcdf-fortran-4.4.5/CMakeLists.txt netcdf-fortran-4.4.4-source/CMakeLists.txt
--- netcdf-fortran-4.4.5/CMakeLists.txt	2016-05-13 15:17:37.000000000 -0600
+++ netcdf-fortran-4.4.5-source/CMakeLists.txt	2017-11-27 11:06:04.000000000 -0700
@@ -555,7 +555,7 @@
   ADD_DEFINITIONS(-DLONGLONG_IS_LONG)
   # for easier debugging of cfortran.h
   IF (APPLE)
-    SET (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fmacro-backtrace-limit=0")
+    #SET (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fmacro-backtrace-limit=0")
   endif ()
 ENDIF()
 
