diff --color -ruN netcdf-c-4.9.3-orig/CMakeLists.txt netcdf-c-4.9.3/CMakeLists.txt
--- netcdf-c-4.9.3-orig/CMakeLists.txt	2025-09-19 15:26:36
+++ netcdf-c-4.9.3/CMakeLists.txt	2025-09-19 15:35:49
@@ -346,6 +346,10 @@
     set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
   endif("${isSystemDir}" STREQUAL "-1")
 
+else()
+  set(CMAKE_SKIP_INSTALL_RPATH ON CACHE BOOL "Turn off for static install." FORCE)
+  set(CMAKE_SKIP_RPATH ON CACHE BOOL "Turn off for static install." FORCE)
+
 endif()
 
 ##
