diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0ef674b..9a1289b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1045,6 +1045,8 @@ IF(NC_FLIBS)
   STRING(REPLACE "-lhdf5::hdf5-static" "-lhdf5" NC_FLIBS ${NC_FLIBS})
   STRING(REPLACE "-lhdf5::hdf5_hl-static" "-lhdf5_hl" NC_FLIBS ${NC_FLIBS})
   STRING(REPLACE "-lnetCDF::netcdf" "-lnetcdf" NC_FLIBS ${NC_FLIBS})
+  STRING(REPLACE "-lMPI::MPI_Fortran" "" NC_FLIBS ${NC_FLIBS})
+  STRING(REPLACE "-lnetcdff_c" "" NC_FLIBS ${NC_FLIBS})
 ENDIF()
 
 STRING(REPLACE ";" " " LINKFLAGS "${LINKFLAGS}")
