diff -ruNbB netcdf-fortran-4.4.4/examples/F90/CMakeLists.txt netcdf-fortran-4.4.4-source/examples/F90/CMakeLists.txt
--- netcdf-fortran-4.4.4/examples/F90/CMakeLists.txt	2016-05-13 15:17:37.000000000 -0600
+++ netcdf-fortran-4.4.4-source/examples/F90/CMakeLists.txt	2018-04-19 16:31:05.747166435 -0600
@@ -8,6 +8,7 @@
 FOREACH(F ${example_F90_tests})
   ADD_EXECUTABLE(${prefix}_${F} ${F}.f90)
   TARGET_LINK_LIBRARIES(${prefix}_${F} netcdff ${NETCDF_C_LIBRARY})
+  SET_TARGET_PROPERTIES(${prefix}_${F} PROPERTIES Fortran_MODULE_DIRECTORY ../../fortran)
   IF(MSVC)
     SET_TARGET_PROPERTIES(${prefix}_${F} 
       PROPERTIES LINK_FLAGS_DEBUG " /NODEFAULTLIB:MSVCRT"
