diff --git a/config/BuildSystem/RDict.py b/config/BuildSystem/RDict.py
index cc90f3e26c5774d54da9934b1f8c88e94a50055c..4387f3369bf54f3ac080f8c46810db6d2fe0cf06 100755
--- a/config/BuildSystem/RDict.py
+++ b/config/BuildSystem/RDict.py
@@ -83,7 +83,6 @@ Arg class, which wraps the usual value.'''
   def __init__(self, parentAddr = None, parentDirectory = None, load = 1, autoShutdown = 1, readonly = False):
     import atexit
     import time
-    import xdrlib
 
     self.logFile         = None
     self.setupLogFile()
@@ -98,8 +97,6 @@ Arg class, which wraps the usual value.'''
     self.isServer        = 0
     self.readonly        = readonly
     self.parentDirectory = parentDirectory
-    self.packer          = xdrlib.Packer()
-    self.unpacker        = xdrlib.Unpacker('')
     self.stopCmd         = pickle.dumps(('stop',))
     self.writeLogLine('Greetings')
     self.connectParent(self.parentAddr, self.parentDirectory)
@@ -110,25 +107,20 @@ Arg class, which wraps the usual value.'''
     return
 
   def __getstate__(self):
-    '''Remove any parent socket object, the XDR translators, and the log file from the dictionary before pickling'''
+    '''Remove any parent socket object, and the log file from the dictionary before pickling'''
     self.writeLogLine('Pickling RDict')
     d = self.__dict__.copy()
     if 'parent'    in d: del d['parent']
     if 'saveTimer' in d: del d['saveTimer']
     if '_setCommandLine' in d: del d['_setCommandLine']
-    del d['packer']
-    del d['unpacker']
     del d['logFile']
     return d
 
   def __setstate__(self, d):
-    '''Reconnect the parent socket object, recreate the XDR translators and reopen the log file after unpickling'''
+    '''Reconnect the parent socket object and reopen the log file after unpickling'''
     self.logFile  = open('RDict.log', 'a')
     self.writeLogLine('Unpickling RDict')
     self.__dict__.update(d)
-    import xdrlib
-    self.packer   = xdrlib.Packer()
-    self.unpacker = xdrlib.Unpacker('')
     self.connectParent(self.parentAddr, self.parentDirectory)
     return
 
@@ -477,13 +469,9 @@ Arg class, which wraps the usual value.'''
       p = packet
     else:
       p = pickle.dumps(packet)
-    self.packer.reset()
-    self.packer.pack_uint(len(p))
     if hasattr(s, 'write'):
-      s.write(self.packer.get_buffer())
       s.write(p)
     else:
-      s.sendall(self.packer.get_buffer())
       s.sendall(p)
     self.writeLogLine(source+': Sent packet')
     return
@@ -495,9 +483,6 @@ Arg class, which wraps the usual value.'''
       s.read(4)
       value = pickle.load(s)
     else:
-      # I probably need to check that it actually read these 4 bytes
-      self.unpacker.reset(s.recv(4))
-      length    = self.unpacker.unpack_uint()
       objString = ''
       while len(objString) < length:
         objString += s.recv(length - len(objString))
diff --git a/config/gmakegen.py b/config/gmakegen.py
index c9c476822c5f26e2eb5543ae91d8635377ae9c7b..a843b94cdcc3117ea5d378e704bb11dabf90ac78 100644
--- a/config/gmakegen.py
+++ b/config/gmakegen.py
@@ -1,7 +1,6 @@
 #!/usr/bin/env python3
 
 import os
-from sysconfig import _parse_makefile as parse_makefile
 import sys
 import logging
 sys.path.insert(0, os.path.abspath(os.path.dirname(__file__)))
@@ -36,6 +35,16 @@ def getlangsplit(name):
 def stripsplit(line):
   return line[len('#requires'):].replace("'","").split()
 
+def parse_makefile(fn, out=None):
+    if out is None:
+        out = {}
+    with open(fn) as f:
+        for l in f:
+            if "=" in l:
+                a,b = l.split("=", 1)
+                out[a.strip()] = b.strip()
+    return out
+
 PetscPKGS = 'sys vec mat dm ksp snes ts tao'.split()
 # the key is actually the language suffix, it won't work for suffixes such as 'kokkos.cxx' so use an _ and replace the _ as needed with .
 LANGS = dict(kokkos_cxx='KOKKOS', hip_cpp='HIP', sycl_cxx='SYCL', raja_cxx='RAJA', c='C', cxx='CXX', cpp='CPP', cu='CU', F='F', F90='F90')
diff --git a/doc/changes/dev.rst b/doc/changes/dev.rst
index c1c338d6ae83b496bb9c237fd8ec6d0ee326cd8c..600dc567c398d6008fff4cb7d4a0e6dc8bfa371e 100644
--- a/doc/changes/dev.rst
+++ b/doc/changes/dev.rst
@@ -14,6 +14,7 @@ Changes: Development
 .. rubric:: Configure/Build:
 
 - Add ``--with-openmp-kernels``
+- Update to work with python-3.13
 
 .. rubric:: Sys:
 
