diff --color -Naru petsc-3.21-orig/config/BuildSystem/config/framework.py petsc-3.21/config/BuildSystem/config/framework.py
--- petsc-3.21-orig/config/BuildSystem/config/framework.py	2025-09-20 10:26:57
+++ petsc-3.21/config/BuildSystem/config/framework.py	2025-09-20 10:29:56
@@ -585,6 +585,8 @@
       lines = [s for s in lines if s.find('skipping incompatible') < 0]
       # Multiple gfortran libraries present
       lines = [s for s in lines if s.find('may conflict with libgfortran') < 0]
+      # Multiple mpi libraries present
+      lines = [s for s in lines if s.find('may conflict with libmpi') < 0]
       # macOS libraries built for different macOS versions
       lines = [s for s in lines if s.find(' was built for newer macOS version') < 0]
       lines = [s for s in lines if s.find(' was built for newer OSX version') < 0]
