diff -Naur petsc-v3.13-old/config/BuildSystem/script.py petsc-v3.13-new/config/BuildSystem/script.py
--- petsc-v3.13-old/config/BuildSystem/script.py	2020-03-29 22:59:40.000000000 -0400
+++ petsc-v3.13-new/config/BuildSystem/script.py	2022-09-11 12:22:22.000000000 -0400
@@ -236,7 +236,7 @@
         thread = InShell()
         thread.start()
         thread.join(timeout)
-        if thread.isAlive():
+        if thread.is_alive():
           error = 'Runaway process exceeded time limit of '+str(timeout)+'\n'
           log.write(error)
           return ('', error, -1)
