--- petsc-3.5.2-source/src/mat/impls/aij/mpi/fdmpiaij.c	2015-06-19 12:18:32.428790761 -0700
+++ petsc-3.5.2-source-new/src/mat/impls/aij/mpi/fdmpiaij.c	2015-06-19 12:19:05.881376148 -0700
@@ -602,7 +602,7 @@ PetscErrorCode MatFDColoringCreate_MPIXA
    bcols is chosen s.t. dy-array takes 50% of memory space as mat */
   ierr = MatGetBlockSize(mat,&bs);CHKERRQ(ierr);
   ierr = PetscObjectTypeCompare((PetscObject)mat,MATMPIBAIJ,&isBAIJ);CHKERRQ(ierr);
-  if (isBAIJ) {
+  if (isBAIJ || m==0) {
     c->brows = m;
     c->bcols = 1;
   } else { /* mpiaij matrix */
