diff -ruNbB petsc-3.8.2/src/mat/impls/hypre/mhypre.c petsc-3.8.2-hypre/src/mat/impls/hypre/mhypre.c
--- petsc-3.8.2/src/mat/impls/hypre/mhypre.c	2017-11-09 09:12:33.000000000 -0700
+++ petsc-3.8.2-hypre/src/mat/impls/hypre/mhypre.c	2018-05-17 21:42:18.000000000 -0600
@@ -1448,6 +1448,6 @@
 static PetscErrorCode hypre_array_destroy(void *ptr)
 {
    PetscFunctionBegin;
-   hypre_TFree(ptr);
+   hypre_TFree(ptr, HYPRE_MEMORY_HOST);
    PetscFunctionReturn(0);
 }
