diff -Naur seacas-2022-02-16-old/packages/seacas/libraries/exodus/src/ex_utils.c seacas-2022-02-16-new/packages/seacas/libraries/exodus/src/ex_utils.c
--- seacas-2022-02-16-old/packages/seacas/libraries/exodus/src/ex_utils.c	2022-02-16 13:22:23.000000000 -0700
+++ seacas-2022-02-16-new/packages/seacas/libraries/exodus/src/ex_utils.c	2025-07-13 23:15:47.088924872 -0600
@@ -1742,7 +1742,10 @@
         */
 
         /* const int NC_SZIP_EC = 4; */ /* Selects entropy coding method for szip. */
-        const int NC_SZIP_NN = 32;      /* Selects nearest neighbor coding method for szip. */
+#if !defined(NC_SZIP_NN)
+        const int NC_SZIP_NN = 32; /* Selects nearest neighbor coding method for szip. */
+#endif
+        /* const int NC_SZIP_NN = 32; Selects nearest neighbor coding method for szip. */
         /* Even and between 4 and 32; typical values are 8, 10, 16, 32 */
         const int SZIP_PIXELS_PER_BLOCK =
             file->compression_level == 0 ? 32 : file->compression_level;
