diff -ruNbB superlu-5.2.1-original/CBLAS/CMakeLists.txt superlu-5.2.1-source/CBLAS/CMakeLists.txt
--- superlu-5.2.1-original/CBLAS/CMakeLists.txt	2016-05-22 09:58:44.000000000 -0600
+++ superlu-5.2.1-source/CBLAS/CMakeLists.txt	2017-10-18 11:21:56.000000000 -0600
@@ -80,3 +80,7 @@
 endif()
 
 add_library(blas ${sources} ${HEADERS})
+if (APPLE)
+  set_target_properties(blas PROPERTIES LINK_FLAGS "-Wl,-undefined,dynamic_lookup")
+endif()
+
diff -ruNbB superlu-5.2.1-original/TESTING/MATGEN/CMakeLists.txt superlu-5.2.1-source/TESTING/MATGEN/CMakeLists.txt
--- superlu-5.2.1-original/TESTING/MATGEN/CMakeLists.txt	2016-05-22 09:58:44.000000000 -0600
+++ superlu-5.2.1-source/TESTING/MATGEN/CMakeLists.txt	2017-10-18 11:22:37.000000000 -0600
@@ -97,3 +97,6 @@
 endif()
 
 add_library(matgen ${sources})
+if (APPLE)
+  set_target_properties(matgen PROPERTIES LINK_FLAGS "-Wl,-undefined,dynamic_lookup")
+endif()
diff -ruNbB superlu-5.2.1-original/make.inc superlu-5.2.1-source/make.inc
--- superlu-5.2.1-original/make.inc	2016-05-22 10:08:28.000000000 -0600
+++ superlu-5.2.1-source/make.inc	2017-10-18 11:17:45.000000000 -0600
@@ -14,31 +14,31 @@
 ############################################################################
 #
 # ToDo: Make this work for shared libraries and out-of-source builds.  Right
-# now it only works for builds under /home/xiaoye/Dropbox/Codes/SuperLU/SuperLU_5.2.1/build/
+# now it only works for builds under /Users/lipnikov/ASCEM/build-tpls/build/tpls/superlu/superlu-5.2.1-source/build/
 #
 #  The name of the libraries to be created/linked to
 #
-SuperLUroot	= /home/xiaoye/Dropbox/Codes/SuperLU/SuperLU_5.2.1/build
-SUPERLULIB   	= $(SuperLUroot)/SRC/libsuperlu.a
+SuperLUroot	= /Users/lipnikov/ASCEM/build-tpls/build/tpls/superlu/superlu-5.2.1-source/build
+SUPERLULIB   	= $(SuperLUroot)/SRC/libsuperlu.so
 
 TMGLIB       	= libtmglib.a
 
 # BLASDEF 	= -DUSE_VENDOR_BLAS
-BLASLIB		= /home/xiaoye/Dropbox/Codes/SuperLU/SuperLU_5.2.1/build/CBLAS/libblas.a
-LIBS		= $(SUPERLULIB) /home/xiaoye/Dropbox/Codes/SuperLU/SuperLU_5.2.1/build/CBLAS/libblas.a
+BLASLIB		= /Users/lipnikov/ASCEM/build-tpls/build/tpls/superlu/superlu-5.2.1-source/build/CBLAS/libblas.so
+LIBS		= $(SUPERLULIB) /Users/lipnikov/ASCEM/build-tpls/build/tpls/superlu/superlu-5.2.1-source/build/CBLAS/libblas.so
 
 #
 #  The archiver and the flag(s) to use when building archive (library)
 #  If your system has no ranlib, set RANLIB = echo.
 #
-ARCH         = /usr/bin/ar
+ARCH         = /opt/local/bin/ar
 ARCHFLAGS    = cr
-RANLIB       = /usr/bin/ranlib
+RANLIB       = /opt/local/bin/ranlib
 
-CC           = /usr/bin/cc
-CFLAGS 	     = -O3 -DNDEBUG -DPRNTlevel=0 -DAdd_  
+CC           = /opt/local/bin/mpicc
+CFLAGS 	     = -O3 -DNDEBUG -DPRNTlevel=0 -DAdd_  -fPIC 
 NOOPTS       = -O0
-FORTRAN	     = /usr/bin/gfortran
+FORTRAN	     = /opt/local/bin/mpif90
 
 LOADER       = $(CC)
 LOADOPTS     =
