--- superlu-7.0.1/CBLAS/CMakeLists.txt	2025-05-07 17:47:22
+++ superlu-7.0.1-patch/CBLAS/CMakeLists.txt	2025-06-26 17:57:46
@@ -82,6 +82,9 @@
 #endif()
 
 add_library(blas ${sources})
+if (APPLE)
+  set_target_properties(blas PROPERTIES LINK_FLAGS "-Wl,-undefined,dynamic_lookup")
+endif()
 
 include(GNUInstallDirs)
 install(TARGETS blas
--- superlu-7.0.1/TESTING/MATGEN/CMakeLists.txt	2025-05-07 17:47:22
+++ superlu-7.0.1-patch/TESTING/MATGEN/CMakeLists.txt	2025-06-26 17:58:59
@@ -102,3 +102,6 @@
 
 add_library(matgen ${sources})
 target_link_libraries(matgen superlu)
if (APPLE)
  set_target_properties(matgen PROPERTIES LINK_FLAGS "-Wl,-undefined,dynamic_lookup")
endif()
