diff -Naur trilinos-11.6.1-Source-old/packages/fei/test_utils/snl_fei_tester.cpp trilinos-11.6.1-Source/packages/fei/test_utils/snl_fei_tester.cpp
--- trilinos-11.6.1-Source-old/packages/fei/test_utils/snl_fei_tester.cpp	2014-01-29 16:06:17.000000000 -0700
+++ trilinos-11.6.1-Source/packages/fei/test_utils/snl_fei_tester.cpp	2016-08-08 23:32:35.000000000 -0600
@@ -523,7 +523,7 @@
   FEI_OFSTREAM outfile(str.c_str());
 
   if (!outfile || outfile.bad()) {
-    fei::console_out() << "ERROR opening solution output file " << fileName << FEI_ENDL;
+    fei::console_out() << "ERROR opening solution output file " << fileName.str() << FEI_ENDL;
     return(-1);
   }
 
@@ -591,7 +591,7 @@
   FEI_OFSTREAM outfile(str.c_str());
 
   if (!outfile || outfile.bad()) {
-    fei::console_out() << "ERROR opening solution output file " << fileName << FEI_ENDL;
+    fei::console_out() << "ERROR opening solution output file " << fileName.str() << FEI_ENDL;
     return(-1);
   }
 
@@ -668,7 +668,7 @@
   FEI_OFSTREAM outfile(str.c_str());
 
   if (!outfile || outfile.bad()) {
-    fei::console_out() << "ERROR opening solution output file " << fileName << FEI_ENDL;
+    fei::console_out() << "ERROR opening solution output file " << fileName.str() << FEI_ENDL;
     return(-1);
   }
 
