diff -rupN trilinos-10.6.4-Source/packages/mesquite/src/ObjectiveFunction/ObjectiveFunction.hpp trilinos-10.6.4-Source-patch/packages/mesquite/src/ObjectiveFunction/ObjectiveFunction.hpp
--- trilinos-10.6.4-Source/packages/mesquite/src/ObjectiveFunction/ObjectiveFunction.hpp	2011-03-23 09:50:22.000000000 -0600
+++ trilinos-10.6.4-Source-patch/packages/mesquite/src/ObjectiveFunction/ObjectiveFunction.hpp	2012-01-20 12:32:37.000000000 -0700
@@ -45,6 +45,9 @@ Header file for the Mesquite::ObjectiveF
 
 #include "Mesquite.hpp"
 
+/* GCC 4.6 Patch */
+#include <cstddef>
+
 #include <list>
 #include <vector>
 
diff -rupN trilinos-10.6.4-Source/packages/mesquite/src/QualityMetric/PMeanPMetric.hpp trilinos-10.6.4-Source-patch/packages/mesquite/src/QualityMetric/PMeanPMetric.hpp
--- trilinos-10.6.4-Source/packages/mesquite/src/QualityMetric/PMeanPMetric.hpp	2011-03-23 09:50:22.000000000 -0600
+++ trilinos-10.6.4-Source-patch/packages/mesquite/src/QualityMetric/PMeanPMetric.hpp	2012-01-20 12:32:18.000000000 -0700
@@ -36,6 +36,9 @@
 #include "Mesquite.hpp"
 #include "Exponent.hpp"
 
+/* GCC 4.6 Patch */
+#include <cstddef>
+
 #include <vector>
 
 namespace MESQUITE_NS {
diff -rupN trilinos-10.6.4-Source/packages/mesquite/src/QualityMetric/TMP/TargetMetricUtil.hpp trilinos-10.6.4-Source-patch/packages/mesquite/src/QualityMetric/TMP/TargetMetricUtil.hpp
--- trilinos-10.6.4-Source/packages/mesquite/src/QualityMetric/TMP/TargetMetricUtil.hpp	2011-03-23 09:50:23.000000000 -0600
+++ trilinos-10.6.4-Source-patch/packages/mesquite/src/QualityMetric/TMP/TargetMetricUtil.hpp	2012-01-20 12:32:00.000000000 -0700
@@ -34,6 +34,10 @@
 #define MSQ_TARGET_METRIC_UTIL_HPP
 
 #include "Mesquite.hpp"
+
+/* GCC 4.6 Patch */
+#include <cstddef>
+
 #include <vector>
 
 namespace MESQUITE_NS {
diff -rupN trilinos-10.6.4-Source/packages/pamgen/src/uns_inline_decomp.h trilinos-10.6.4-Source-patch/packages/pamgen/src/uns_inline_decomp.h
--- trilinos-10.6.4-Source/packages/pamgen/src/uns_inline_decomp.h	2011-03-23 09:50:36.000000000 -0600
+++ trilinos-10.6.4-Source-patch/packages/pamgen/src/uns_inline_decomp.h	2012-01-20 12:31:26.000000000 -0700
@@ -1,6 +1,9 @@
 #ifndef uns_inline_decompH
 #define uns_inline_decompH
 
+/* GCC 4.6 Patch */
+#include <cstddef>
+
 #include <list>
 #include <vector>
 #include "topology_enum.h"
Binary files trilinos-10.6.4-Source/packages/stk/stk_util/stk_util/environment/.RuntimeMessage.hpp.swp and trilinos-10.6.4-Source-patch/packages/stk/stk_util/stk_util/environment/.RuntimeMessage.hpp.swp differ
diff -rupN trilinos-10.6.4-Source/packages/stk/stk_util/stk_util/environment/RuntimeMessage.hpp trilinos-10.6.4-Source-patch/packages/stk/stk_util/stk_util/environment/RuntimeMessage.hpp
--- trilinos-10.6.4-Source/packages/stk/stk_util/stk_util/environment/RuntimeMessage.hpp	2011-03-23 09:50:46.000000000 -0600
+++ trilinos-10.6.4-Source-patch/packages/stk/stk_util/stk_util/environment/RuntimeMessage.hpp	2012-01-23 08:35:07.000000000 -0700
@@ -9,6 +9,9 @@
 #ifndef STK_UTIL_ENVIRONMENT_RUNTIMEMESSAGE_HPP
 #define STK_UTIL_ENVIRONMENT_RUNTIMEMESSAGE_HPP
 
+/* GCC 4.6 Patch */
+#include <cstddef>
+
 #include <iosfwd>
 #include <vector>
 
diff -rupN trilinos-10.6.4-Source/packages/teuchos/src/Teuchos_ConfigDefs.hpp trilinos-10.6.4-Source-patch/packages/teuchos/src/Teuchos_ConfigDefs.hpp
--- trilinos-10.6.4-Source/packages/teuchos/src/Teuchos_ConfigDefs.hpp	2011-03-23 09:50:56.000000000 -0600
+++ trilinos-10.6.4-Source-patch/packages/teuchos/src/Teuchos_ConfigDefs.hpp	2012-01-20 12:31:37.000000000 -0700
@@ -60,6 +60,9 @@
 #  define HAVE_COMPLEX
 #endif
 
+/* GCC 4.6 Patch */
+#include <cstddef>
+
 #include <cstdio>
 #include <cstdarg>
 #include <cerrno>
