diff -r 4e258311568d src/alquimia/fortran/pflotran_alquimia_interface.F90
--- a/src/alquimia/fortran/pflotran_alquimia_interface.F90	Wed Feb 05 14:55:49 2014 -0800
+++ b/src/alquimia/fortran/pflotran_alquimia_interface.F90	Wed Jun 18 15:42:33 2014 -0700
@@ -366,7 +366,7 @@
   ! process the condition
   !
   call  c_f_string_ptr(condition%name, name)
-  write (*, '(a, a)') "processing : ", trim(name)
+  !write (*, '(a, a)') "processing : ", trim(name)
 
   if (condition%aqueous_constraints%size > 0) then
      ! the driver is supplying the constraint data, so we need to
@@ -380,7 +380,7 @@
   else
      ! the driver just supplied a name, so we check for a constraint
      ! with that name in the pflotran input file and use that.
-     write (*, '(a, a)') "Looking for pflotran constraint : ", trim(name)
+     !write (*, '(a, a)') "Looking for pflotran constraint : ", trim(name)
      tran_constraint => engine_state%transport_constraints%first
      do
         if (associated(tran_constraint)) then
