diff --git a/cmake/TPLs/FindTPLSuperLUDist.cmake b/cmake/TPLs/FindTPLSuperLUDist.cmake
index 233e173b3d0..c18b9a933de 100644
--- a/cmake/TPLs/FindTPLSuperLUDist.cmake
+++ b/cmake/TPLs/FindTPLSuperLUDist.cmake
@@ -62,6 +62,7 @@ TRIBITS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES( SuperLUDist
 include(CheckCSourceCompiles)
 include(MultilineSet)
 
+
 # Versions 3.0 and later of SuperLU_DIST namespace the IterRefine_t
 # enum values with "SLU_" (e.g. "SLU_DOUBLE" versus "DOUBLE" in
 # previous versions).  Check which style is used so that packages like
@@ -82,9 +83,10 @@ int main()
   SET(CMAKE_REQUIRED_INCLUDES ${TPL_SuperLUDist_INCLUDE_DIRS})
   SET(CMAKE_REQUIRED_LIBRARIES ${TPL_SuperLUDist_LIBRARIES} ${TPL_ParMETIS_LIBRARIES} ${TPL_METIS_LIBRARIES} ${TPL_BLAS_LIBRARIES})
   SET(CMAKE_REQUIRED_FLAGS ${CMAKE_EXE_LINKER_FLAGS})
-  CHECK_C_SOURCE_COMPILES("${SOURCE}" ${VARNAME})
+  CHECK_CXX_SOURCE_COMPILES("${SOURCE}" ${VARNAME})
 ENDFUNCTION()
 
+
 # Version 4.0 of SuperLU_DIST changed the calling parameters of the
 # LUstructInit function.  Check which is used here.
 FUNCTION(CHECK_SUPERLUDIST_LUSTRUCTINIT  VARNAME)
@@ -104,7 +106,7 @@ int main()
   SET(CMAKE_REQUIRED_INCLUDES ${TPL_SuperLUDist_INCLUDE_DIRS})
   SET(CMAKE_REQUIRED_LIBRARIES ${TPL_SuperLUDist_LIBRARIES} ${TPL_ParMETIS_LIBRARIES} ${TPL_METIS_LIBRARIES} ${TPL_BLAS_LIBRARIES})
   SET(CMAKE_REQUIRED_FLAGS ${CMAKE_EXE_LINKER_FLAGS})
-  CHECK_C_SOURCE_COMPILES("${SOURCE}" ${VARNAME})
+  CHECK_CXX_SOURCE_COMPILES("${SOURCE}" ${VARNAME})
 ENDFUNCTION()
 
 IF (TPL_ENABLE_SuperLUDist)